nvflare/lighter/poc_commands.py

```diff
@@ -120,28 +152,72 @@ def is_poc_ready(poc_workspace: str):
 
 def validate_poc_workspace(poc_workspace: str):
     if not is_poc_ready(poc_workspace):
-        print(f"workspace {poc_workspace} is not ready, please use poc --prepare to prepare poc workspace")
-        sys.exit(2)
+        raise Exception(f"workspace {poc_workspace} is not ready, please use poc --prepare to prepare poc workspace")
+
 
+def validate_gpu_ids(gpu_ids: list, host_gpu_ids: list):
+    for gpu_id in gpu_ids:
+        if gpu_id not in host_gpu_ids:
+            raise ValueError(f"gpu_id provided is not available in the host machine, available GPUs are {host_gpu_ids}")
+
+
+def get_host_gpu_ids() -> List[int]:
+    process = subprocess.Popen(["nvidia-smi", "--list-gpus"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    result = process.communicate()
+    st = process.poll()
+    rc = process.returncode
+    if rc > 0:
+        raise Exception("Failed to get host gpu device Ids", result[0])
+    else:
+        # 'GPU 0: NVIDIA GeForce RTX 3090 (UUID: GPU-xxxx-xxxx-xxxx-xxx)\n'
+        if result[0].startswith("GPU"):
+            gpus = result[0].split("\n")
+            gpu_ids = [int(gpu.split(":")[0].split(" ")[1]) for gpu in gpus[:-1]]
+        else:
+            gpu_ids = []
+
+    return gpu_ids
+
+
+def get_gpu_ids(user_input_gpu_ids, host_gpu_ids) -> List[int]:
+    if type(user_input_gpu_ids) == int and user_input_gpu_ids < 0:
+        gpu_ids = host_gpu_ids
+    else:
+        gpu_ids = user_input_gpu_ids
+        validate_gpu_ids(gpu_ids, host_gpu_ids)
+    return gpu_ids
 
-def start_poc(poc_workspace: str, white_list: list = []):
-    print(f"start_poc at {poc_workspace}, white_list={white_list}")
+
+def start_poc(poc_workspace: str, gpu_ids: List[int], white_list: list = []):
+    print(f"start_poc at {poc_workspace}, gpu_ids={gpu_ids}, white_list={white_list}")
     validate_poc_workspace(poc_workspace)
-    _run_poc(SC.CMD_START, poc_workspace, excluded=[SC.FLARE_OVERSEER], white_list=white_list)
+    _run_poc(SC.CMD_START, poc_workspace, gpu_ids, excluded=[SC.FLARE_OVERSEER], white_list=white_list)
 
 
 def stop_poc(poc_workspace: str, white_list: list = []):
     print(f"stop_poc at {poc_workspace}")
     validate_poc_workspace(poc_workspace)
-    _run_poc(SC.CMD_STOP, poc_workspace, excluded=[SC.FLARE_OVERSEER], white_list=white_list)
+    gpu_ids: List[int] = []
+    _run_poc(SC.CMD_STOP, poc_workspace, gpu_ids, excluded=[SC.FLARE_OVERSEER], white_list=white_list)
+
+
+def _get_clients(package_commands: list) -> List[str]:
+    clients = [
+        package_dir_name
+        for package_dir_name, _ in package_commands
+        if package_dir_name != SC.FLARE_CONSOLE
+           and package_dir_name != SC.FLARE_SERVER
+           and package_dir_name != SC.FLARE_OVERSEER
```

Comment:
POC should not have overseer, maybe we should also remove all overseer related stuff.